extensions/wikia/Email/EmailController.class.php

```diff
@@ -425,6 +497,30 @@ private function generateMobileApplicationsBadges() {
 		return $badges;
 	}
 
+	protected function getSocialIconsList($prefix = null) {
+		$lang = $this->targetLang;
+		$socials = EmailController::SOCIALS;
+		if(!isset($socials[$lang])) {
+			$lang = 'en';
+		}
+		$socialIcons = EmailController::SOCIALS[$lang];
+		$result = [];
+		foreach ($socialIcons as $socialName) {
+			$socialIconName = $socialName;
+			if(!is_null($prefix)) {
+				$socialIconName = $prefix . $socialName;
+			}
+			$icon = ImageHelper::$icons[$socialIconName];
+			$result[] = [
+				'iconSrc' => ImageHelper::getFileUrl($icon['name'] . '.' .  $icon['extension']),
+				'iconAlt' => $socialName,
+				'iconUrl' => EmailController::SOCIAL_URLS[$socialName][$lang],
+				'iconClass' => strtolower($socialName)
+			];
+		}
+		return $result;
+	}
+
 	/**
 	 * TODO Move this into unsubscribe extension?
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this "TODO" comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1135)